.pre-commit-config.yaml

```diff
@@ -18,6 +18,32 @@ repos:
     language: system
     files: >-
       ^setup\.py$
+  - id: pip-compile
+    name: pip-compile
+    entry: pip-compile -q --no-annotate --output-file=test-requirements.txt setup.py test-requirements.in
+    language: system
+    files: ^(test-requirements\.(txt|in)|)$
+    pass_filenames: false
+  - id: pip-compile-docs
+    name: pip-compile-docs
+    entry: pip-compile -q --no-annotate --output-file=docs/requirements.txt docs/requirements.in setup.py
+    language: system
+    files: ^(docs\/requirements\.(txt|in)|)$
+    pass_filenames: false
+  - id: pip-compile-upgrade
+    name: pip-compile-upgrade
+    entry: pip-compile -q --upgrade --no-annotate --output-file=test-requirements.txt setup.py test-requirements.in
+    language: system
+    files: ^(test-requirements\.(txt|in)|)$
+    pass_filenames: false
+    stages: [manual]
+  - id: pip-compile-docs-upgrade
+    name: pip-compile-docs-upgrade
+    entry: pip-compile -q --upgrade --no-annotate --output-file=docs/requirements.txt docs/requirements.in setup.py
+    language: system
+    files: ^(docs\/requirements\.(txt|in)|)$
+    pass_filenames: false
+    stages: [manual]
```

Comment:
@webknjaz I am not really proud of the duplication here but I was not able to find a better way so far. If you ever find a better approach let me know.